### PR TITLE
[IOTDB-6194] Remove _list from parameter target_config_node_list

### PR DESCRIPTION
**Pull request by CRZbulabula**
https://issues.apache.org/jira/browse/IOTDB-6194

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java [hunk 1]
```diff
@@ -168,11 +168,16 @@ private void loadProperties(Properties properties) throws BadNodeUrlException, I
...
+    String seedConfigNode = properties.getProperty(IoTDBConstant.CN_SEED_CONFIG_NODE, null);
+    if (seedConfigNode == null) {
+      seedConfigNode = properties.getProperty(IoTDBConstant.CN_TARGET_CONFIG_NODE_LIST, null);
+      LOGGER.warn(
+          "The parameter cn_target_config_node_list has been abandoned, "
+              + "only the first ConfigNode address will be used to join in the cluster. "
+              + "Please use cn_seed_config_node instead.");
+    }
+    if (seedConfigNode != null) {
+      conf.setSeedConfigNode(NodeUrlUtils.parseTEndPointUrls(seedConfigNode.trim()).get(0));
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java:180]: move to line 177?

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java [hunk 2]
```diff
@@ -168,11 +168,16 @@ private void loadProperties(Properties properties) throws BadNodeUrlException, I
...
-      conf.setTargetConfigNode(NodeUrlUtils.parseTEndPointUrl(targetConfigNodes.trim()));
+    String seedConfigNode = properties.getProperty(IoTDBConstant.CN_SEED_CONFIG_NODE, null);
+    if (seedConfigNode == null) {
+      seedConfigNode = properties.getProperty(IoTDBConstant.CN_TARGET_CONFIG_NODE_LIST, null);
+      LOGGER.warn(
+          "The parameter cn_target_config_node_list has been abandoned, "
+              + "only the first ConfigNode address will be used to join in the cluster. "
+              + "Please use cn_seed_config_node instead.");
+    }
+    if (seedConfigNode != null) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java:179]: else?

File: iotdb-core/datanode/src/test/resources/datanode3conf/iotdb-datanode.properties [hunk 3]
```diff
@@ -26,7 +26,7 @@ dn_mpp_data_exchange_port=10742
 dn_schema_region_consensus_port=10752
 dn_data_region_consensus_port=10762
 
-dn_target_config_node_list=127.0.0.1:10710,127.0.0.1:10711,127.0.0.1:10712
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/test/resources/datanode3conf/iotdb-datanode.properties:29]: use one node endpoint?

File: iotdb-core/datanode/src/test/resources/datanode2conf/iotdb-datanode.properties [hunk 4]
```diff
@@ -26,7 +26,7 @@ dn_mpp_data_exchange_port=10741
 dn_schema_region_consensus_port=10751
 dn_data_region_consensus_port=10761
 
-dn_target_config_node_list=127.0.0.1:10710,127.0.0.1:10711,127.0.0.1:10712
+dn_seed_config_node=127.0.0.1:10710,127.0.0.1:10711,127.0.0.1:10712
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/test/resources/datanode2conf/iotdb-datanode.properties:29]: use one node endpoint?

File: iotdb-core/datanode/src/test/resources/datanode1conf/iotdb-datanode.properties [hunk 5]
```diff
@@ -26,7 +26,7 @@ dn_mpp_data_exchange_port=10740
 dn_schema_region_consensus_port=10750
 dn_data_region_consensus_port=10760
 
-dn_target_config_node_list=127.0.0.1:10710,127.0.0.1:10711,127.0.0.1:10712
+dn_seed_config_node=127.0.0.1:10710,127.0.0.1:10711,127.0.0.1:10712
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/test/resources/datanode1conf/iotdb-datanode.properties:29]: use one node endpoint?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java [hunk 6]
```diff
@@ -1993,14 +1993,20 @@ private void loadCQProps(Properties properties) {
...
-    String configNodeUrls = properties.getProperty(IoTDBConstant.DN_TARGET_CONFIG_NODE_LIST);
+    String configNodeUrls = properties.getProperty(IoTDBConstant.DN_SEED_CONFIG_NODE);
+    if (configNodeUrls == null) {
+      configNodeUrls = properties.getProperty(IoTDBConstant.DN_TARGET_CONFIG_NODE_LIST);
+      logger.warn(
+          "The parameter dn_target_config_node_list has been abandoned, "
+              + "only the first ConfigNode address will be used to join in the cluster. "
+              + "Please use dn_seed_config_node instead.");
+    }
     if (configNodeUrls != null) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:2004]: else?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java [hunk 7]
```diff
@@ -1993,14 +1993,20 @@ private void loadCQProps(Properties properties) {
...
       try {
         configNodeUrls = configNodeUrls.trim();
-        conf.setTargetConfigNodeList(NodeUrlUtils.parseTEndPointUrls(configNodeUrls));
+        conf.setSeedConfigNode(NodeUrlUtils.parseTEndPointUrls(configNodeUrls));
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:2007]: move to line 2003 and use the first one?

Comment (CRZbulabula) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:2007]: Line 1998 for compability

File: iotdb-core/datanode/src/test/resources/datanode1conf/iotdb-datanode.properties (hunk 5 above)
Comment (CRZbulabula) [reply iotdb-core/datanode/src/test/resources/datanode1conf/iotdb-datanode.properties]: Updated

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java (hunk 6 above)
Comment (CRZbulabula) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:2004]: Line 1998 for compability

File: iotdb-core/datanode/src/test/resources/datanode2conf/iotdb-datanode.properties (hunk 4 above)
Comment (CRZbulabula) [reply iotdb-core/datanode/src/test/resources/datanode2conf/iotdb-datanode.properties]: Updated

File: iotdb-core/datanode/src/test/resources/datanode3conf/iotdb-datanode.properties (hunk 3 above)
Comment (CRZbulabula) [reply iotdb-core/datanode/src/test/resources/datanode3conf/iotdb-datanode.properties:29]: Updated

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java (hunk 2 above)
Comment (CRZbulabula) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java:179]: The same as before

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java (hunk 1 above)
Comment (CRZbulabula) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java:180]: These lines should be here to keep compability.

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java [hunk 8]
```diff
@@ -127,7 +127,7 @@ public void removeConfigNode(TConfigNodeLocation removedNode)
   }
 
   /**
-   * config_node_list of confignode-system.properties
+   * config_node of confignode-system.properties
```
[the remark this answers is not in the document]
Comment (CRZbulabula) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java]: My mistake.

File: iotdb-core/confignode/src/assembly/resources/conf/iotdb-confignode.properties [hunk 9]
```diff
@@ -38,12 +38,12 @@ cn_consensus_port=10720
 ### Target Config Nodes
 ####################
 
-# For the first ConfigNode to start, cn_target_config_node_list points to its own cn_internal_address:cn_internal_port.
-# For other ConfigNodes that to join the cluster, cn_target_config_node_list points to any running ConfigNode's cn_internal_address:cn_internal_port.
+# For the first ConfigNode to start, cn_target_config_node points to its own cn_internal_address:cn_internal_port.
+# For other ConfigNodes that to join the cluster, cn_target_config_node points to any running ConfigNode's cn_internal_address:cn_internal_port.
 # Note: Currently only endpoints that point to a single confignode are supported
```
[the remark this answers is not in the document]
Comment (CRZbulabula) [reply iotdb-core/confignode/src/assembly/resources/conf/iotdb-confignode.properties]: Updated

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java (hunk 7 above)
Comment (CRZbulabula) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:2007]: Fixed

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java (hunk 1 above)
Comment (CRZbulabula) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeDescriptor.java:180]: Fixed

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**